menu/reader.js

```diff
@@ -0,0 +1,42 @@
+const fs = require("fs");
+const YAML = require("js-yaml");
+
+class Menu {
+    constructor(text) {
+        this.delivery = text['Delivery'];
+        this.instructions = text['Instructions']; // If instructions provided
+        this.dishes = text['Menu'];
+        this.modifiers = text['Modifiers'];
+    }
+
+}
+
+// Convert .yml file to yml text
+function readYmlFile(filename) {
+    try {
+        const raw = fs.readFileSync(filename, 'utf-8');
+        return raw
+    } catch (e) {
+        throw new Error("File cannot be Read!");
+    }
+}
+
+// Returns menu object
+function getMenu(rawText) {
+    try {
+        const data = YAML.load(rawText)
+        const menu = new Menu(data);
+        return menu;
+    } catch (e) {
+        throw new Error("YML parsing Error!");
```

Comment:
Yep I tested and it seems to catch the errors separately. The functions are executed sequentially since I'm reading the file to raw text and then I am parsing that raw text so in that case the two exceptions are caught separately.